core/editor/src/tools/ellipse.rs

```diff
@@ -34,6 +34,7 @@ impl Default for EllipseToolFsmState {
 #[derive(Clone, Debug, Default)]
 struct EllipseToolData {
 	drag_start: ViewportPosition,
+	index: u64,
```

Comment:
Is `index` just a hack to deal with unique naming? This should be able to go away after switching from a hashmap to vector approach.